### ISSUE TITLE
C# Editor, 'finally' blocks not foldable

### ISSUE THREAD

**Issue by vsfeedback** [dotnet/roslyn#39797]
_This issue has been moved from [a ticket on Developer Community](https://developercommunity.visualstudio.com/content/idea/795282/c-editor-finally-blocks-not-foldable.html)._

---
```
finally
{
  // foo
}
```
Expected: Foldable (what Microsoft calls &quot;Outline&quot;)
Actual: Not foldable.

Feature Suggestion:
In Microsoft lingo: Enhance the User Experience by making User not Experience non-Outlinable source code blocks when encountering `finally`. Instead make User Experience the Enabling of the Feature Experience of Experiencing `finally` become Outlinable. The synergi effects of Buzzword Experience, while Experiencing the Outlining Experience could become an Enhancing Experience in Experiencing the Enhanced Outlining Experience for this keyword.

In laymans terms: Make `finally` foldable.

---
#### Visual Studio Feedback System on 10/28/2019, 01:29 AM: 

Thank you for taking the time to provide your suggestion. We will do some preliminary checks to make sure we can proceed further.  We'll provide an update once the issue has been triaged by the product team.

**Comment by sharwell** [comment]
This is closely related to #28058

**Comment by mavasani** [comment]
Design meeting notes: Existing feature supports outlining for if/else but not for try/catch/finally 

**Conclusion:** Mark as up for grabs to add support

**Comment by Dreamescaper** [comment]
Currently `if` and `else` blocks are separate, meaning that folding `if` leaves `else` block unfolded.
`try-catch-finally` currently is considered as single block, meaning that folding `try` block folds `catch-finally` as well.
Should it be changed to have them separate as well, i.e. folding `try` should leave `catch-finally` unfolded?

**Comment by sharwell** [comment]
> Should it be changed to have them separate as well, i.e. folding try should leave catch-finally unfolded?

Yes, `try`, `catch`, and `finally` would fold individually instead of all together.

**Comment by KirillOsenkov** [comment]
Would be really nice to have outlining for `catch` and `finally`.
